plugin/manager.go

```diff
@@ -0,0 +1,124 @@
+package plugin
+
+import (
+	"fmt"
+	"github.com/cloudquery/cloudquery/logging"
+	"github.com/hashicorp/go-plugin"
+	"github.com/rs/zerolog/log"
+	"os"
+	"os/exec"
+	"sync"
+)
+
+var (
+	doOnce   sync.Once
+	instance *Manager
+)
+
+type Manager struct {
+	activeClients map[string]*plugin.Client
```

Comment:
Looks like there is a bit of duplication here (I know it was also in my code but if we are already refactoring...). This also causes duplicate add and remove in setup and cleanup (also found some minor bug in cleanup of registry).

I suggest `registry map[string]struct{
    client *plugin.Client
    provider CQProvider
}`